src/main/java/net/dries007/tfc/common/blockentities/ToolRackBlockEntity.java

```diff
@@ -90,4 +89,19 @@ private void insertItem(int slot, ItemStack stack)
         level.playSound(null, worldPosition, TFCSounds.TOOL_RACK_PLACE.get(), SoundSource.BLOCKS, 1, 1 + ((level.random.nextFloat() - level.random.nextFloat()) / 16));
     }
 
+    // Sorry for this.
+    // This returns a unique analog value for every permutation of items within slots
+    @Override
+    public int getAnalogValue()
```

Comment:
This just seems unintuitive to me, imo. Yes, in theory, people could use this to detect which element in a tool rack is present, but it makes much more sense to just output a count of the number of items in the tool rack.